### ISSUE TITLE
FK to existing object is ignored, SubFactory is used instead

### ISSUE THREAD

**Issue by grokcode** [FactoryBoy/factory_boy#571]
#### Description
When using `SQLAlchemyModelFactory`,  if a foreign key to an existing database record is provided, the FK is ignored, a new record is created via the SubFactory, and the provided FK is overwritten with the FK of the record created via the SubFactory.

#### To Reproduce
Using the below factories and models, `AddressFactory()` works as expected, creating a new `Address` and a `Person` via the SubFactory and committing them to the db. 

However, `AddressFactory(person_id=1)`, where 1 is the id of a person that already exists in the db, does not work as I was expecting it to. I'm thinking that in this case, an Address should be created with a foreign key to the existing Person. Instead, a new Address is created, a new Person is created via the SubFactory, and the Address is set to use the new Person.

##### Model / Factory code
```python
## factories.py
from factory.alchemy import SQLAlchemyModelFactory
from factory import Faker, SubFactory


class PersonFactory(SQLAlchemyModelFactory):
    class Meta:
        sqlalchemy_session = db.session
        sqlalchemy_session_persistence = 'commit'
        model = models.Person

    first_name = Faker('first_name')


class AddressFactory(SQLAlchemyModelFactory):
    class Meta:
        sqlalchemy_session = db.session
        sqlalchemy_session_persistence = 'commit'
        model = models.Address

    person = SubFactory(PersonFactory)


## models.py
from sqlalchemy import Column, String, Integer, ForeignKey
from sqlalchemy.orm import relationship


class Person:
    id = Column(Integer, primary_key=True)
    first_name = Column(String(25))

    addresses = relationship('Address', backref="person")

class Address:
    person_id = Column(Integer, ForeignKey('person.id', ondelete='CASCADE'))
```

#### Notes
Possibly related to #463?

**Comment by rbarrois** [comment]
It is kinda related to #463 and #69; however, in your declaration, there is no way for factory_boy to know that a relation exists between ``person`` and ``person_id``.
When you call ``AddressFactory(person_id=1)``, you end up with:

```python
person = Person.create(**params)
Address.create(
    person=person,
    person_id=1,
)
```

If you want some kind of ``get_or_create`` behaviour, we'd need to add it to ``SQLAlchemyFactory; see [the code in ``DjangoModelFactory`` for inspiration](https://github.com/FactoryBoy/factory_boy/blob/master/factory/django.py#L135-L166).
If you did add this to ``SQLAlchemyFactory`` (or a custom subclass of it) and declared the ``id`` field to be part of said ``get_or_create``, you would get the expected behaviour :-)

**Comment by francoisfreitag** [comment]
Closing due to inactivity.

`sqlalchemy_get_or_create` was implemented in 59ee6584d8788af065a86aa168e8f42a4e9f3801 and is now being superseded by #794. Feel free to reopen an issue if you think the original concern was not resolved.
